test/Escrow.t.sol

```diff
@@ -273,6 +274,101 @@ contract EscrowContractTest is Test {
+    }
+
+    function testAssignDeposit() public {
+        depositUnassignedEscrow(SENDER_ADDRESS, SENDER_ADDRESS, ESCROW_AMOUNT);
+
+        uint256 depositedAmount = escrowContract.getEscrowAmount(SENDER_ADDRESS, receiversAddresses[0]);
+        assertEq(depositedAmount, ZERO_AMOUNT, "Incorrect deposited amount");
+
+        assignDeposit(SENDER_ADDRESS, receiversAddresses[0], ESCROW_AMOUNT);
+
+        depositedAmount = escrowContract.getEscrowAmount(SENDER_ADDRESS, receiversAddresses[0]);
+        assertEq(depositedAmount, ESCROW_AMOUNT, "Incorrect deposited amount"); 
+    }
+
+    function testAssignDepositWithInsufficientBalance() public {
+        depositUnassignedEscrow(SENDER_ADDRESS, SENDER_ADDRESS, ESCROW_AMOUNT);
+
+        uint256 depositedAmount = escrowContract.getEscrowAmount(SENDER_ADDRESS, receiversAddresses[0]);
+        assertEq(depositedAmount, ZERO_AMOUNT, "Incorrect deposited amount");
+
+        vm.prank(SENDER_ADDRESS);
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                Escrow.InsufficientUnassignedEscrow.selector, 
+                ESCROW_AMOUNT, 
+                ESCROW_AMOUNT * 2
+            )
+        );
+        escrowContract.assignDeposit(receiversAddresses[0], ESCROW_AMOUNT * 2);
+    }
+
+    function testAssignDepositMany() public {
+        uint256[] memory amounts = new uint256[](3);
+        amounts[0] = ESCROW_AMOUNT;
+        amounts[1] = ESCROW_AMOUNT*10;
+        amounts[2] = ESCROW_AMOUNT*2;
+
+        uint256 totalAmount = 0;
+        for (uint i = 0; i < amounts.length; i++) {
+           totalAmount += amounts[i]; 
+        }
+
+        depositUnassignedEscrow(SENDER_ADDRESS, SENDER_ADDRESS, totalAmount);
+
+        assignDepositMany(SENDER_ADDRESS, receiversAddresses, amounts);
+
+        vm.prank(SENDER_ADDRESS);
+
+        for (uint i = 0; i < receiversAddresses.length; i++) {
+            uint256 depositedAmount = escrowContract.getEscrowAmount(SENDER_ADDRESS, receiversAddresses[i]);
+            assertEq(depositedAmount, amounts[i], "Incorrect deposited amount");
+        }
+    }
+
+    function testAssignManyFundsWithLengthMismatch() public {
+        uint256[] memory amounts = new uint256[](2);
+        amounts[0] = ESCROW_AMOUNT;
+        amounts[1] = ESCROW_AMOUNT;
+
+        vm.prank(SENDER_ADDRESS);
+        vm.expectRevert(Escrow.InputsLengthMismatch.selector);
+        escrowContract.assignDepositMany(receiversAddresses, amounts);
+    }
+
+    function testAssignDepositManyWithInsuficientBalance() public {
+        uint256[] memory amounts = new uint256[](3);
+        amounts[0] = ESCROW_AMOUNT;
+        amounts[1] = ESCROW_AMOUNT*10;
+        amounts[2] = ESCROW_AMOUNT*2;
+
+        uint256 totalAmount = 0;
+        for (uint i = 0; i < amounts.length; i++) {
+           totalAmount += amounts[i]; 
+        }
+
+        depositUnassignedEscrow(SENDER_ADDRESS, SENDER_ADDRESS, ESCROW_AMOUNT);
```

Comment:
Might be good to test depositing `totalAmount-1` to test for any off-by-one regressions